src/tr2/game/lara/misc.c

```diff
@@ -1803,3 +1803,14 @@ void Lara_Extinguish(void)
         effect_num = next_effect_num;
     }
 }
+
+bool Lara_IsM16Active(void)
+{
+    if (g_Lara.weapon_item == NO_ITEM || g_Lara.gun_type != LGT_M16) {
+        return false;
+    }
+
+    const ITEM *const item = Item_Get(g_Lara.weapon_item);
+    return item->current_anim_state == 0 || item->current_anim_state == 2
+        || item->current_anim_state == 4;
+}
```

Comment:
Can we add a description to what this function does?

Reply:
Thanks, done. Added it to the header. It's sort of an awkward function but as the code was duplicated in four places, I felt it warranted separating out.